src/cpp/shared/combat/army.cpp

```diff
@@ -1803,6 +2134,6 @@ void army::InitClean() {
   this->mirrorIdx = -1;
   this->armyIdx = -1;
   this->previousQuantity = -1;
-  if(CreatureHasAttribute(this->creatureIdx, ASTRAL_DODGE))
+  if(CreatureHasAttribute(this->creatureIdx, ASTRAL_DODGE) || CreatureHasAttribute(this->creatureIdx, JUMPER))
```

Comment:
abilityCounter is cleared every turn for the assassin, but not for the berserker. These work differently and represent different things. They should probably use different fields.

Also, I can't comment on it, but: As long as you're touching this method, can you use NELEM(this->combatSounds) instead of 7 on line 2121?

Reply:
It's still a counter for ability uses, so I thought it's logical to reuse the same field.
I didn't understand the NELEM thing you mentioned

Reply:
NELEM: I meant this: https://github.com/jkoppel/project-ironfist/blob/master/src/cpp/shared/base.h#L12

Reply:
How would you implement a creature that both has jump and shadow dodge?

Reply:
Well then that's a more general problem since current code doesn't support more than 1 ironfist ability properly

Reply:
"Well then that's a more general problem since current code doesn't support more than 1 ironfist ability properly"

Wait, what? It should be trivial to give the assassins strike-and-return, right?

Reply:
Strike-and-return is not using the counter. I meant those abilities that use a counter

Reply:
Current code doesn't support more than one ability that uses the counter, because you designed it in a way so that only one ability can use the counter.

Change the design.